hadoop-hdds/common/src/main/java/org/apache/hadoop/util/GenericCheckedSupplier.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.util;
+
+/**
+ * Similar to {@link java.util.function.Supplier}, this class presents a block
+ * of code generating a value with a possibility of throwing a generic
+ * Exception.
+ */
+
+@FunctionalInterface
+public interface GenericCheckedSupplier<T, E extends Exception> {
```

Comment:
Hi @duongkame, thank you for the review! I tried implementing this change, but looks like it would require modifying 15-20+ files to update the method signatures to throw `Exception` instead of a specific checked exception. 

This more general type of exception (i.e. `Exception`) at multiple method signatures can potentially lead to more ambiguous error handling.

Reply:
Since `captureLatencyNs` method only declares `IOException` as a checked exception, in order to handle the `OS3Exception` thrown by the `createS3Bucket` method (for instance), the `catch` block can catch the exception and throw a `RuntimeException` instead. 

I think this could be done as an alternative because `RuntimeException` is an unchecked exception and does not need to be explicitly declared in the `throws` clause of the method signature. By throwing a `RuntimeException`, we avoid the need to modify the signature of the `captureLatencyNs` method to include `OS3Exception` in its throws clause (the import was causing a circular dependency on the modules) and it can instead be handled at a higher level in the call stack. 

Please let me know your thoughts on the same, thanks.

Reply:
Hi, I'm ok either way. The metric capture can be simpler I think. Did a [quick test](https://github.com/apache/ozone/commit/5bd40008e96e132533a918eef8657bf701af95bb).